pkg/ruler/compat.go

```diff
@@ -51,11 +51,9 @@ func (a *appendableAppender) AddFast(l labels.Labels, ref uint64, t int64, v flo
 }
 
 func (a *appendableAppender) Commit() error {
-	if _, err := a.pusher.Push(a.ctx, client.ToWriteRequest(a.samples)); err != nil {
-		return err
-	}
+	_, err := a.pusher.Push(a.ctx, client.ToWriteRequest(a.samples))
```

Comment:
The change in behaviour here is named in the commit message, but not motivated, and not mentioned in the pull request.

I'd like to know why this needs to change?  Why is it related to the other changes in this pull request?

Reply:
It used to be that we would create a new appender for each rule group evaluation, so samples that failed to be pushed were cleared. Now that the appender is longer lived it needs to clear failed samples, otherwise each request will have all the old samples and the new samples. I saw that behavior quickly overwhelming an ingester.